File: src/test/java/leetcode/TrieTest.java
package leetcode;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class TrieTest {
  private Trie trie;

  @BeforeEach
  void setUp() {
    trie = new Trie();
  }

  @Test
  void insert() {
    trie.insert("apple");
    assertThat(trie.search("apple")).isTrue();
  }

  @Test
  void search() {
    trie.insert("apple");
    assertThat(trie.search("apple")).isTrue();
    assertThat(trie.search("app")).isFalse();
  }

  @Test
  void startsWith() {
    trie.insert("apple");
    assertThat(trie.startsWith("app")).isTrue();
  }

  @Test
  void trie() {
    trie.insert("apple");

    assertThat(trie.search("apple")).isTrue();
    assertThat(trie.search("app")).isFalse();
    assertThat(trie.startsWith("app")).isTrue();

    trie.insert("app");

    assertThat(trie.search("app")).isTrue();
  }
}

File: leetcode-kt/src/test/kotlin/com/giwankim/leetcode/TrieTest.kt
package com.giwankim.leetcode

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class TrieTest :
    FunSpec({
        lateinit var sut: Trie

        beforeTest {
            sut = Trie()
        }

        context("trie") {
            test("insert") {
                sut.insert("apple")
                sut.search("apple") shouldBe true
            }

            test("search") {
                sut.insert("apple")
                sut.search("apple") shouldBe true
                sut.search("app") shouldBe false
            }

            test("startsWith") {
                sut.insert("apple")
                sut.startsWith("app") shouldBe true
            }

            test("trie") {
                sut.insert("apple")

                sut.search("apple") shouldBe true
                sut.search("app") shouldBe false
                sut.startsWith("app") shouldBe true

                sut.insert("app")

                sut.search("app") shouldBe true
            }
        }
    })
